bin/check_samplesheet.py

```diff
@@ -43,37 +49,37 @@ def __init__(
         Args:
             sample_col (str): The name of the column that contains the sample name
                 (default "sample").
-            first_col (str): The name of the column that contains the first (or only)
-                FASTQ file path (default "fastq_1").
-            second_col (str): The name of the column that contains the second (if any)
-                FASTQ file path (default "fastq_2").
-            single_col (str): The name of the new column that will be inserted and
-                records whether the sample contains single- or paired-end sequencing
-                reads (default "single_end").
+            type_col (str): The name of the column that contains the dataype for
+                the read data (default "datatype").
+            file_col (str): The name of the column that contains the file path for
+                the read data (default "datafile").
+            index_col (str): The name of the column that contains the index file
+                for the data (default "indexfile").
 
         """
         super().__init__(**kwargs)
+
         self._sample_col = sample_col
-        self._first_col = first_col
-        self._second_col = second_col
-        self._single_col = single_col
+        self._type_col = type_col
+        self._file_col = file_col
+        self._index_col = index_col
         self._seen = set()
         self.modified = []
 
     def validate_and_transform(self, row):
         """
-        Perform all validations on the given row and insert the read pairing status.
+        Perform all validations on the given row.
 
         Args:
             row (dict): A mapping from column headers (keys) to elements of that row
                 (values).
 
         """
         self._validate_sample(row)
-        self._validate_first(row)
-        self._validate_second(row)
-        self._validate_pair(row)
-        self._seen.add((row[self._sample_col], row[self._first_col]))
+        self._validate_type(row)
+        self._validate_data_file(row)
+        self._validate_index_file(row)
+        self._seen.add((row[self._sample_col], row[self._file_col], row[self._index_col]))
```

Comment:
It doesn't make sense to have the same sample and BAM file twice but with different BAI. I would restrict this to just  the sample and the data file

Reply:
index removed